### ISSUE TITLE
Set enum member on instruction with constant value

### ISSUE THREAD

**Issue by fergofrog** [Vector35/binaryninja-api#1216]
Often when comparisons against enum members (or ```#define``` constants) are compiled they become static ```mov r, val``` where val is the constant. Currently binary ninja allows enum members to be set against data, but not when in code. Implementing this would make for a much tidier disassembly, all the way up to the MLIL.

**Comment by plafosse** [comment]
Thanks for the feature request.

**Comment by xusheng6** [comment]
Related to https://github.com/Vector35/binaryninja-api/issues/2606

**Comment by xusheng6** [comment]
I tentatively suggest that we reopen issues like https://github.com/Vector35/binaryninja-api/issues/3211. That one is tracking a different issue than this one. This issue tracks the ability to manually specify an integer constant as an enum type, while #3211 seems to suggest we handle the enum better _automatically_. For example, we should be able to change the constant 2 to an enum name without user interaction in the following code:

```
100003ed8      enum foobar var_28
100003ed8      if (var_28 == 2)
100003ee8          _printf("foo3\n")
```

**Comment by xusheng6** [comment]
Related to #1703

**Comment by xusheng6** [comment]
This should be converted into several sub-tasks:

- [ ] Do better type resolving/propagation on enums and figure out more constant values are enums automatically when possible
- [ ] Provide a UI and API to mark a selected constant value as an enum type. The UI should try to automatically find all enum members across all known enums with the constant's value

**Comment by xusheng6** [comment]
The second task is blocked on #3490

**Comment by cxnder** [comment]
Resolved in builds >= 3.2.3974
